### ISSUE TITLE
Aqara RTCGQ13LM

### ISSUE THREAD

**Issue by u20p17** [dresden-elektronik/deconz-rest-plugin#5109]
## Device

- Product name: Aqara RTCGQ13LM high precision motion sensor
- Manufacturer: Aqara 
- Model identifier: RTCGQ13LM 
- Device type : 
  - Sensor

## Screenshots

![overview](https://user-images.githubusercontent.com/63282286/125100982-fb164d80-e0d9-11eb-9a6f-d0846b42c6fa.JPG)
![Node_Info](https://user-images.githubusercontent.com/63282286/125100947-f356a900-e0d9-11eb-8493-bfdb89161985.JPG)

### Basic
![Basic](https://user-images.githubusercontent.com/63282286/125101969-16ce2380-e0db-11eb-9e28-ae57d3d3f105.JPG)

### Identify
![Identify](https://user-images.githubusercontent.com/63282286/125101105-2305b100-e0da-11eb-9e2f-4875b091249a.JPG)

### Other clusters that are not mentioned above
Occupancy sensing:
![Occupancy_sensing](https://user-images.githubusercontent.com/63282286/125101155-331d9080-e0da-11eb-9dcf-69423ca4a5a2.JPG)
power configuration:
![power_configuration](https://user-images.githubusercontent.com/63282286/125101177-387adb00-e0da-11eb-9162-a74a5fae273e.JPG)
OTAU:
![OTAU](https://user-images.githubusercontent.com/63282286/125101340-66f8b600-e0da-11eb-9f54-3e5407c6ec19.JPG)

**Comment by SwoopX** [comment]
Can you please update occupancy sensing and power configuration cluster screenshots? The values still contain the defaults; you must use the read button and make sure the sensor is awake by triggering its function 2-3 secs prior to reading.

Before you do that, can you please update your firmware? I hope you're using a Conbee II 😉 
[deCONZ_ConBeeII_0x266f0700.bin.zip](https://github.com/dresden-elektronik/deconz-rest-plugin/files/6795994/deCONZ_ConBeeII_0x266f0700.bin.zip)
It has some special magic included which should tickle even more out of the device. After firmware update, please reset and re-pair the device.

**Comment by u20p17** [comment]
Hi,
thanks for your answer. I now updated from 0x266d0700 to 0x266f0700. After that I deleted the node and addedit again --> after that I couldn't read in the values with my HomeAssistant deconz running... I now tried to connect the Conbee2 stick to my laptop and there it worked.
node info:
![node info](https://user-images.githubusercontent.com/63282286/125185007-60755600-e222-11eb-967a-475e39254399.JPG)

Basic:
![basic](https://user-images.githubusercontent.com/63282286/125184988-450a4b00-e222-11eb-84cb-36b3a37454d2.JPG)

identify:
![identify](https://user-images.githubusercontent.com/63282286/125184992-4f2c4980-e222-11eb-8e8f-a46713d510cf.JPG)

LUMI_specific:
![LUMI_specific_1](https://user-images.githubusercontent.com/63282286/125184998-55bac100-e222-11eb-9347-3cc6fd989150.JPG)
![LUMI_specific_2](https://user-images.githubusercontent.com/63282286/125185000-581d1b00-e222-11eb-9e41-6e939c777f3d.JPG)

occupancy:
![occupancy](https://user-images.githubusercontent.com/63282286/125185009-679c6400-e222-11eb-9386-84a9aacf12e5.JPG)

OTAU:
![OTAU_1](https://user-images.githubusercontent.com/63282286/125185014-6d924500-e222-11eb-84cb-986a323e1117.JPG)
![OTAU2](https://user-images.githubusercontent.com/63282286/125185015-6f5c0880-e222-11eb-9a82-8dda021576cf.JPG)

power_configuration:
![power_configuration](https://user-images.githubusercontent.com/63282286/125185018-7420bc80-e222-11eb-9704-722480f478f6.JPG)


BR

**Comment by SwoopX** [comment]
Great, that helps. Thanks! You you please get me some further info?

For the power configuration cluster, please double click on attribute's 0x0020 value and read the reporting configuration (at the bottom). Do the same for occupancy sensing cluster on attribute 0x0010. I'd like to stick to the defaults here if they're reasonable...

Would you be able to compile the plugin to test the changes?

**Comment by u20p17** [comment]
Hi,
good to hear. :) For 0x0020  of the power configuration cluster it worked, but for 0x0010 in occupancy sensing cluster I always get 'reading config failed'.

![power_reporting_config](https://user-images.githubusercontent.com/63282286/125234845-cd035a00-e2e1-11eb-99cc-648743490840.JPG)
![occupancy_reporting_config](https://user-images.githubusercontent.com/63282286/125234866-d42a6800-e2e1-11eb-996f-0f010cdd529d.JPG)

Regarding compilation: I am not the expert in this so I would prefer, if you could do that for me if possible :)

**Comment by SwoopX** [comment]
Thanks again. So the reporting for power values are a little bit weak, will set them to report once an hour. The occupancy sensing it a bit odd, that's neither an "I support it" nor an "I don't support it". I guess there'll be no harm anyway, so I wouldn't bother too much.

> Regarding compilation: I am not the expert in this so I would prefer, if you could do that for me if possible :)

As I don't have the device, that's kinda pointless, although I'd love to 😉

**Comment by u20p17** [comment]
Thank you for your effort! Can you give ma a short description how I could compile it? With some help from you I should be able to do that :)

**Comment by u20p17** [comment]
Hi @SwoopX , not sure if you are waiting for me to compile and test a new version for this or if you are working on this topic and will come back to me when you have finished? (Please do not get me wrong, I don't have that big time pressure with the new sensors and do not want to put you under stress... just that I know when I should do some tests for you again :) )

**Comment by SwoopX** [comment]
All good. I'm just not sure how to cater with setting the sensitivity on the sensor. Was hoping for a better answer than I got from the other devs 😉 

Once this is sorted out, I'll give you some guidance on how to compile and test, given that you don't use HA.

**Comment by u20p17** [comment]
Hi, thank you very much!

Just du be clear: I am using HomeAssistant with the Conbee2 stick for my system. :)

**Comment by L3H0** [comment]
@u20p17 Hi, can you tell me where you purchase this device?

**Comment by u20p17** [comment]
Hi @SwoopX ,
today I updated my conbee 2 firmware to 26700700 and tried to connect to this sensor with Phoscon, but I wasn't able to connect... Do I also have to update phoscon? If noit, how could I generate a log file which you might need for analyses?

BR

**Comment by u20p17** [comment]
Hi @SwoopX ,
After a while I had time again to test the new sensors with HA. Therefore I updated to v6.10.0 (changelog: Bump deCONZ to 2.12.6) and the Conbee2 Stick to 26700700. Then I tried to add the sensors with Phoscon, but without success. I was not able to add them. Furthermore I checked if they get displayed on deconz and since the update I do not get any additionb node there (like with the old software version). What could be the problem? If you need any log file please let me know.
BR

**Comment by SwoopX** [comment]
I'd assume you do not correctly reset the device. Also the selected snipped is not really helpful since it's just 9 seconds. What you see there has nothing to do with the motion sensor.

I don't know HA, so when it comes to any related question, I cannot be of any help.

**Comment by u20p17** [comment]
ok, I will try again to reset the sensor. Furthermore I could also try to set up another raspi with deconz running on it (without HA) and check if it is working there... 

Regarding the log: what would be helpful for you in order to check the problem (if the same also appears with the secound raspbi...)

Thanks!

**Comment by 7leman7** [comment]
![image](https://user-images.githubusercontent.com/80063876/135862042-ddf64f62-cdec-4557-9dcc-f15b73108887.png)

I can confirm I was able to delete and successfully add back RTCGQ13LM sensor in deCONZ 2.12.6 running as HA addon. I can finally see the sensitivity settings etc. Will see how it works (before that - when added as generic sensor - it was acting weirdly).

I also have ConBee II, FW 26680700. I was adding the sensor via Phoscon app (it does not appear in Phoscon, only in deCONZ).

**Comment by u20p17** [comment]
@7leman7 : Good to hear that at least you can add the sensor in deconz. I now tried it again, but I am not able to do it. Maybe you can describe what you did?

i did it now the following:
1.) removed the batteries for half a minute.
2.) inserted it again and pressed the button in the rear very long (until the blue light started flashing.)
3.) opened Phoscon Menu --> sensors and there I pressed add new sensor an as manufacturer i have chosen 'others'.
4.) after that I pressed again the button on the back until it flashed again but in phoscon it couldn't add it.
5.) After that I opened deconz and also there I can not see the node anymore.

I now also tried a secound (completely new) sensor and there the same is happening.

@SwoopX: Do you have an explanation why @7leman7 can see it in deconz (and if i understand correctly, the sensor is working like expected with the change in sensitivity) but it is not possible for him to add the sensor to phoscon?

**Comment by 7leman7** [comment]
I did not remove batteries, just deleted the respective node in deconz.

After that I restarted HA, opened Phoscon, started sensor add process (as "other"), pressed the button on the sensor for 5 seconds (sensor blinked 3 times), waited until Phoscon showed add process status as "ready", pressed the button - that's it.

Sensor does not show in Phoscon, but in deCONZ I get new node "Presence **" where ** is the node number. I understand this is the expected behavior.

**Comment by 7leman7** [comment]
BTW, from what I can see during the last hour the sensor works OK.

**Comment by u20p17** [comment]
ok, I now tried with the latest firmware(26720700) and with the 26680700, but with the same result:
![Unbenannt](https://user-images.githubusercontent.com/63282286/135886523-08e1b813-4e8f-474b-a6bd-5dadb179b878.JPG)

I can not add it in phoscon nor can I see it in deconz. If I look at the debug view can just see the following ( If you check this @7leman7  do you also get those messages? (you can display it by pressing help-debugview)

![Unbenannt2](https://user-images.githubusercontent.com/63282286/135886554-5d4cc459-24e0-4d76-a42d-e9a88e94fc16.JPG)

**Comment by SwoopX** [comment]
You seem to have more than one Phoscon tab open accross all devices at your place. That interferes with opening the network properly.

**Comment by u20p17** [comment]
I now closed all phoscon and deconz tabs and then tried to connect again. Same behaviour than in my last post (one hour ago)... 

I update the conbee2 stick always on my windows pc and the reconnect it to the pi + rebooting the whole system - could this somehow be the problem?

**Comment by 7leman7** [comment]
> ok, I now tried with the latest firmware(26720700) and with the 26680700, but with the same result: ![Unbenannt](https://user-images.githubusercontent.com/63282286/135886523-08e1b813-4e8f-474b-a6bd-5dadb179b878.JPG)
> 
> I can not add it in phoscon nor can I see it in deconz. If I look at the debug view can just see the following ( If you check this @7leman7 do you also get those messages? (you can display it by pressing help-debugview)
> 
> ![Unbenannt2](https://user-images.githubusercontent.com/63282286/135886554-5d4cc459-24e0-4d76-a42d-e9a88e94fc16.JPG)

I don't have the "unhandled APS-DATA" lines, other than that my log looks pretty similar, but I don't really know what I'm looking at :)

Also, I don't have the inclusion timeframe in the logs already (rebooted HA since then).

**Comment by u20p17** [comment]
i now found out that all the unhandled APS-DATA error come from a hue smart plug - when I disconnect it and delete the node no errors occur anymore... 
Furthermore I tried to use the conbee 2 with 26720700 and 2.12.06 on my windows pc and there I can connect the sensor and it get displayed as node in Deconz ( in Phoscon I cannot see it - so the same as for @7leman7 - why isn't it also displayed in phoscon as a motion sensor?). Then I deleted the node on the win pc again and connected the conbee2 to my raspberry pi with the same software versions running. Here I can not connect the sensor via phoscon... strange behaviour...

**Comment by u20p17** [comment]
so now i was also able to connect the sensor with deconz and phoscon running in HA on my pi. it had the conbee2 connected with an elero centero Transmitter Stick on one usb hub and after disconnecting the elero stick it weas working - maybe the conbee didn't get enough power? However this is working now.

Now only one question is left for me: Why isn't it also displayed in phoscon as a motion sensor? If this is indented, how can I change the name of the node in deconz?

**Comment by 7leman7** [comment]
Yeah, same for me: after reading "PIR Occupied To Unoccupied Delay" shows 120, config call via rest-api shows duration 60 (also changed to 60 myself after joining the sensor). Maybe I was wring above, but I can say fir sure I did my change via deCONZ, not via rest-api.

Really unsure what's going on, in the meanwhile I was bale to get back all clusters (by numerous "read simple descriptor" calls) except for lumi specific cluster...

**Comment by 7leman7** [comment]
Okay, so after numerous deletions and inclusions I finally managed to pair the sensor with Lumi Specific Cluster available again.

With that I can change sensitivity attribute in Lumi Specific Cluster and change PIR Occupied to Unoccupied delay to 60 in Occupancy Sensing cluster (writing works, reading back gives 60 as just written).

What I did was press the pairing button for 5 sec (until 3 blinks) and and then pressing the button once a second until paired ("Done" message appearing in Phoscon "Add sensor" menu). Still the process seems to be a bit flimsy as Phoscon showed the sensor to be successfully paired even though it apparently did not recognise it correctly during several previous attempts.
